packages/perseus/src/widgets/interactive-graphs/graphs/components/segment.tsx

```diff
@@ -1,44 +1,136 @@
-import {vec, useMovable} from "mafs";
-import {useRef} from "react";
+import {useMovable, vec} from "mafs";
+import {useRef, useState} from "react";
 import * as React from "react";
 
 import useGraphConfig from "../../reducer/use-graph-config";
-import {TARGET_SIZE} from "../../utils";
+import {snap, TARGET_SIZE} from "../../utils";
 import {useTransformVectorsToPixels} from "../use-transform";
 import {getIntersectionOfRayWithBox} from "../utils";
 
+import {MovablePointView} from "./movable-point-view";
 import {SVGLine} from "./svg-line";
 import {Vector} from "./vector";
 
 import type {Interval} from "mafs";
 
-const defaultStroke = "var(--movable-line-stroke-color)";
-
 type Props = {
-    start: vec.Vector2;
-    end: vec.Vector2;
-    onMove: (delta: vec.Vector2) => unknown;
-    stroke?: string;
+    points: Readonly<[vec.Vector2, vec.Vector2]>;
+    onMovePoint: (endpointIndex: number, destination: vec.Vector2) => unknown;
+    onMoveLine: (delta: vec.Vector2) => unknown;
+    color?: string;
     /* Extends the line to the edge of the graph with an arrow */
     extend?: {
         start: boolean;
         end: boolean;
-        range: [Interval, Interval];
     };
 };
 
-export const MovableLine = (props: Props) => {
+export const Segment = (props: Props) => {
+    const {
+        onMoveLine,
+        color,
+        points: [start, end],
+        extend,
+    } = props;
+
+    const {visiblePoint: visiblePoint1, focusableHandle: focusableHandle1} =
+        useControlPoint(start, color, (p) => props.onMovePoint(0, p));
+    const {visiblePoint: visiblePoint2, focusableHandle: focusableHandle2} =
+        useControlPoint(end, color, (p) => props.onMovePoint(1, p));
+
+    return (
+        <>
```

Comment:
Beautiful! For posterity, could you add a comment here explaining that the focusHandles/visiblePoints are done this way because of the weirdness with SVGs and z-index/tabindex?